app/search_engines/bento_search/catalog_engine.rb

```diff
@@ -64,8 +64,13 @@ def search_implementation(args)
       holdings = JSON.parse(holdings_response.body)
 
       # TODO: Get a hidden reference to this value into the atom payload so it is referencable from the summary variable
-      if Oga.parse_html(entry['summary']).at_xpath('//a/@href').present?
-        online_resource[Oga.parse_html(entry['summary']).at_xpath('//a/@href').text] = Oga.parse_html(entry['summary']).at_xpath('//a/text()').text.strip
+      parsed_summary = Oga.parse_html(entry['summary'])
+      link_href = parsed_summary.at_xpath('//a/@href')
+      if link_href
+        # NOTE: this was changed mainly to avoid NPE and avoid parsing the same html 3 times. I can't vouch for the
+        # logic (specifically, it feels fragile to assume a correspondence between href and associated text,
+        # but perhaps that assumption is valid in practice?)
+        online_resource[link_href.text] = parsed_summary.at_xpath('//a/text()')&.text&.strip.presence || '[no link text]'
```

Comment:
This works, but I think it could be more simply put:
```
      link = parsed_summary.at_xpath('//a')
      if link
        href = link.attribute('href').to_s
        online_resource[href] = link.text.strip
      end
```

A `link` will _always_ have `.text`. `.attribute('href')` will return `nil` if there's no `href`.

Reply:
Thanks @mksndz! This is cleaner all-around, and addresses (to a reasonable extent, at least) my concern about href/text correspondence. Just pushed changes that incorporate this suggestion.